homeassistant/components/youless/sensor.py

```diff
@@ -0,0 +1,213 @@
+"""The sensor entity for the Youless integration."""
+from datetime import timedelta
+from typing import Any, Dict, Optional
+
+from youless_api import YoulessAPI
+from youless_api.youless_sensor import YoulessSensor
+
+from homeassistant import core
+from homeassistant.components.youless import DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE, DEVICE_CLASS_POWER
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import StateType
+from homeassistant.util import Throttle
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant, entry: ConfigEntry, async_add_entities
+):
+    """Initialize the integration."""
+    gateway = hass.data[DOMAIN][entry.entry_id]
+    device = entry.data[CONF_DEVICE]
+    sensors = [
+        GasSensor(gateway, device),
+        PowerMeterSensor(gateway, device, "low"),
+        PowerMeterSensor(gateway, device, "high"),
+        PowerMeterSensor(gateway, device, "total"),
+        CurrentPowerSensor(gateway, device),
+        DeliveryMeterSensor(gateway, device, "total"),
+        DeliveryMeterSensor(gateway, device, "current"),
+    ]
+
+    async_add_entities(sensors)
+
+
+class YoulessBaseSensor(Entity):
+    """The base sensor for Youless."""
+
+    def __init__(
+        self, gateway: YoulessAPI, device: str, friendly_name: str, sensor_id: str
+    ):
+        """Create the sensor."""
+        self._gateway = gateway
+        self._friendly_name = friendly_name
+        self._device = device
+        self._sensor_id = sensor_id
+
+    @Throttle(timedelta(seconds=30))
```

Comment:
All sensors use the same update... and the throttle is now needed to prevent them from all doing the same thing.

Instead; use the `DataUpdateCoordinator`, see: <https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities>

Reply:
I've added the use of the DataUpdateCoordinator. I hope I implemented it correctly.